rtCommon/bidsArchive.py

```diff
@@ -0,0 +1,825 @@
+        spatialUnits2 = xyztUnits2 % 8
+        spatialMatch = np.array_equal(spatialUnits1, spatialUnits2)
+
+        temporalUnits1 = xyztUnits1 - spatialUnits1
+        temporalUnits2 = xyztUnits2 - spatialUnits2
+        temporalMatch = np.array_equal(temporalUnits1, temporalUnits2)
+
+        # Append compatible if both dimensions match
+        if spatialMatch and temporalMatch:
+            pass
+        # Append compatible if spatial dimensions match, temporal dimensions
+        # don't, and at least one temporal dimension is undefined
+        elif (spatialMatch and not temporalMatch and
+              (temporalUnits1 == 0 or temporalUnits2 == 0)):
+            pass
+        # In any other case, not append compatible
+        else:
+            errorMsg = (
+                f"NIfTI headers not append compatible due to mismatch in "
+                f"xyzt_units field (spatial match: {spatialMatch}, "
+                f"temporal match: {temporalMatch}. "
+                f"xyzt_units 1: {xyztUnits1} | xyzt_units 2: {xyztUnits2}")
+            return (False, errorMsg)
+
+        return (True, "")
+
+    @staticmethod
+    def _metadataAppendCompatible(meta1: dict, meta2: dict) -> Tuple[bool, str]:
+        """
+        Verifies two metadata dictionaries match in a set of required fields. If
+        a field is present in only one or neither of the two dictionaries, this
+        is considered a match.
+
+        This is primarily intended as a safety check, and does not conclusively
+        determine that two images are valid to append to together or are part of
+        the same series.
+
+        Args:
+            meta1: First metadata dictionary
+            meta2: Second metadata dictionary
+
+        Returns:
+            True if all keys that are present in both dictionaries have
+            equivalent values, False otherwise.
+
+        """
+        matchFields = ["Modality", "MagneticFieldStrength", "ImagingFrequency",
+                       "Manufacturer", "ManufacturersModelName",
+                       "InstitutionName", "InstitutionAddress",
+                       "DeviceSerialNumber", "StationName", "BodyPartExamined",
+                       "PatientPosition", "EchoTime",
+                       "ProcedureStepDescription", "SoftwareVersions",
+                       "MRAcquisitionType", "SeriesDescription", "ProtocolName",
+                       "ScanningSequence", "SequenceVariant", "ScanOptions",
+                       "SequenceName", "SpacingBetweenSlices", "SliceThickness",
+                       "ImageType", "RepetitionTime", "PhaseEncodingDirection",
+                       "FlipAngle", "InPlanePhaseEncodingDirectionDICOM",
+                       "ImageOrientationPatientDICOM", "PartialFourier"]
+
+        # If either field is None, skip and continue checking other fields
+        for field in matchFields:
+            value1 = meta1.get(field, None)
+            if value1 is None:
+                continue
```

Comment:
GW comment: what if one value is None and the other is not, should that still pass?